File: Java/AR_237.java
import java.util.Scanner;

public class AR_237 {
    
    static boolean poliDivisible(long num, int tamano) {
        boolean ret;
        
        if      (tamano==1)     ret=true;
        else if (num%tamano==0) ret=poliDivisible(num/10, tamano-1);
        else                    ret=false;
        
        return ret;
    }

    static boolean solve(Scanner scanner) {
        if (!scanner.hasNextLong()) return false;
        
        long num     =scanner.nextLong();               
        String numero=Long.toString(num);
        
        boolean poli=poliDivisible(num, numero.length());        
        if (poli) System.out.println("POLIDIVISIBLE");
        else      System.out.println("NO POLIDIVISIBLE");
        
        return true;
    }

    public static void main(String[] args) {
        Scanner scanner=new Scanner(System.in);
        while (solve(scanner));

        scanner.close();
    }
}

File: C/200-300/AR_237.c
#include <stdio.h>
#include <string.h>

int poliDivisible(long long int num, int tamano) {
    int ret;
    
    if 		(tamano == 1) 		ret=1;
    else if (num % tamano == 0) ret=poliDivisible(num/10, tamano-1);
    else 						ret=0;
    
    return ret;
}

int solve() {
    long long int num;
    if (scanf("%lld", &num)!=1) return 0;
    
    char numero[20];
    sprintf(numero, "%lld", num);
    int size=strlen(numero);

    int poli=poliDivisible(num, size);    
    if (poli) printf("POLIDIVISIBLE\n");
    else 	  printf("NO POLIDIVISIBLE\n");
    
    return 1;
}

int main() {
    while (solve());
	
    return 0;
}